### PR TITLE
Sync battle engine v2

### PR DESCRIPTION
**Pull request by ExpoSeed**
i can't believe i fucked this up 3 times now

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
The only thing changed was the destination branch due to user error, my previous comments on PR #6 are still relevant.  This merge has been tested with a full QA session from the beginning of the game to the Hall of Fame. No obvious issues found.

**Review by AsparagusEduardo — DISMISSED** [review]
Same review as #6 :

> I've done tests to check if current issues are still present after the merge and they still are, so we can at least be sure that the merge is good in that sense.
As this is the first merge in a long time, if there are issues that might originate from this, we can deal with them as they come up.

File: src/battle_util.c [hunk 1]
```diff
@@ -32,8 +31,655 @@
...
+extern const u8 *const gBattlescriptsForBallThrow[];
+extern const u8 *const gBattlescriptsForRunningByItem[];
+extern const u8 *const gBattlescriptsForUsingItem[];
+extern const u8 *const gBattlescriptsForSafariActions[];
+
+static const u8 sPkblToEscapeFactor[][3] = {{0, 0, 0}, {3, 5, 0}, {2, 3, 0}, {1, 2, 0}, {1, 1, 0}};
+static const u8 sGoNearCounterToCatchFactor[] = {4, 3, 2, 1};
+static const u8 sGoNearCounterToEscapeFactor[] = {4, 4, 4, 4};
+
+void HandleAction_UseMove(void)
```
Comment (ultima-soul) [inline src/battle_util.c:55]: Sync these functions based on the gist provided on Discord. The functions were modified before being moved to battle_util.

**Review by ultima-soul — DISMISSED** [review]

**Comment by ExpoSeed** [comment]
Done

**Review by ultima-soul — APPROVED** [review]

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**